src/vapm_collector_recorder.py

```diff
@@ -0,0 +1,86 @@
+
+import os
+import queue
+import shutil
+
+
+import sounddevice as sd
+import soundfile as sf
+
+import numpy
+
+
+
+from vapm_collector_interface import VAPMInterface
+
+
+class VAPMCollectorRecorder(VAPMInterface):
+    FILE_NAME = 'sound'
+    EXTENSION_WAV = '.wav'
+    STORE_PATH = './tmp/recorder/'
+
+    def __init__(self, tmp_path, channels, device, samplerate):
+        super().__init__(tmp_path)
+        self.file_wav = tmp_path
+
+        self.queue_for_store = queue.Queue()
+
+        self.channel = channels
+        self.device = device
+        self.sample_rate = samplerate
+
+
+    def callback(self, indata, frames, time, status):
+        if status:
+            print(status)
+        self.queue_for_store.put(indata.copy())
+
+    def archive_tmp(self, target):
+        print("\n\n Record End:"+repr(self.file_wav))
+        target_filepath = os.path.abspath(target)
+        shutil.copyfile(self.tmp_filepath, target_filepath)
+
+    def _do_collect(self):
+        with sf.SoundFile(self.file_wav, mode='x', samplerate=self.sample_rate,
+                          channels=self.channel, subtype=None) as save_file, sd.InputStream(
+                              samplerate=self.sample_rate, device=self.device,
+                              channels=self.channel, callback=self.callback):
+            print('recording ... ')
+            while True:
+                save_file.write(self.queue_for_store.get())
+
+
+if __name__ == '__main__':
+    import argparse
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        '-t', '--tmp-path', type=str,
```

Comment:
code indentation 잡아야 함: https://github.com/google/styleguide/blob/gh-pages/pyguide.md

```suggestion
    parser.add_argument('-t', '--tmp-path', type=str,
                        default='./tmp/recorder/sound.wav',
                        help='tmp file location')
```
마지막 의미없는 ',' 제거